harness/determined/exec/prep_container.py

```diff
@@ -20,13 +20,20 @@
 from determined.common.api import bindings, certs
 
 
-def trial_prep(sess: api.Session, info: det.ClusterInfo) -> None:
-    trial_info = det.TrialInfo._from_env()
-    trial_info._to_file()
+def download_context_directory(sess: api.Session, info: det.ClusterInfo) -> None:
+    cluster_info = det.ClusterInfo._from_env()
```

Comment:
Could this info be passed with the `info` parameter instead of read fresh from the environment? It always makes me uncomfortable to see globals injected into code -- especially when they're not typed.

Reply:
I think what this comes down to is that some parameters are passed via flags, and some via environment variables, and it's a little hard to track as a reader. There's only so much that you can do about that here right now, but I think that this is a request to read/parse the various parameters in one place. Just like `download_context_directory` isn't the right place to parse flags, it's also probably not the right place to parse parameters read from the environment.

Reply:
I moved the "```trial_info._from_env``` to ```trial_info._to_file```" parsing to where we originally parse ```info```

I was also doing a double parse of ```ClusterInfo```, so this seems a lot better

Reply:
Nice. I think this looks really good, too.